src/CPUSolver.cpp

```diff
@@ -819,14 +819,36 @@ void CPUSolver::transferBoundaryFlux(int track_id,
   /* For the "forward" direction */
   if (direction) {
     start = _tracks[track_id]->isNextOut() * _polar_times_groups;
-    bc = std::min((int)_tracks[track_id]->getBCOut(), 1);
+
+    /* If the outgoing boundary condition is vacuum, set bc to 0 indicating the
+     * track flux of the outgoing track will be zeroed out */
```

Comment:
I guess I thought that the if-else would be in the TrackGenerator's BC initialization code. It would be nice to avoid the if-else here since it is inside the core computationally expensive piece of the MOC algorithm. lbeit, I would not expect this to degrade performance in any substantial way, so leave it as you see fit.